### PR TITLE
Fix: npx webpack make: *** [Makefile:699: public/js/index.js] Error -…

### PR DESCRIPTION
**Pull request by Run2948**
when I execute `TAGS="bindata" make build` to build gitea, it shows:
**npx webpack make: *** [Makefile:699: public/js/index.js] Error -1073741819**
```bash
$ webpack -v
webpack 5.11.1
webpack-cli 4.3.1

$ TAGS="bindata" make build
rm -rf public/js public/css public/fonts public/img/webpack public/serviceworker.js
npx webpack
[webpack-cli] Failed to load 'E:\C_Language\Workspaces\GoProjects\gitea\webpack.config.js' config
[webpack-cli] file:///E:/C_Language/Workspaces/GoProjects/gitea/webpack.config.js:9
import {VueLoaderPlugin} from 'vue-loader';
        ^^^^^^^^^^^^^^^
SyntaxError: The requested module 'vue-loader' is expected to be of type CommonJS, which does not support named exports. CommonJS modules can be imported by importing the default export.
For example:
import pkg from 'vue-loader';
const {VueLoaderPlugin} = pkg;
    at ModuleJob._instantiate (internal/modules/esm/module_job.js:98:21)
    at async ModuleJob.run (internal/modules/esm/module_job.js:137:5)
    at async Loader.import (internal/modules/esm/loader.js:165:24)
    at async loadConfig (E:\C_Language\Workspaces\GoProjects\gitea\node_modules\webpack-cli\lib\webpack-cli.js:1346:35)
    at async WebpackCLI.resolveConfig (E:\C_Language\Workspaces\GoProjects\gitea\node_modules\webpack-cli\lib\webpack-cli.js:1454:38)
    at async WebpackCLI.createCompiler (E:\C_Language\Workspaces\GoProjects\gitea\node_modules\webpack-cli\lib\webpack-cli.js:1839:22)
    at async WebpackCLI.buildCommand (E:\C_Language\Workspaces\GoProjects\gitea\node_modules\webpack-cli\lib\webpack-cli.js:1954:20)
    at async Command.<anonymous> (E:\C_Language\Workspaces\GoProjects\gitea\node_modules\webpack-cli\lib\webpack-cli.js:735:25)
    at async Promise.all (index 1)
    at async Command.<anonymous> (E:\C_Language\Workspaces\GoProjects\gitea\node_modules\webpack-cli\lib\webpack-cli.js:1284:13)
make: *** [Makefile:699: public/js/index.js] Error 2
```

### CODE REVIEW

**Comment by silverwind** [comment]
Ah, this could be a issue with Node's ESM resolution. Wich node version were you encountering this? Perhaps 12? Because I think we only tested this on 14.

File: webpack.config.js
Comment (silverwind) [inline webpack.config.js]:
```suggestion
const {VueLoaderPlugin} = VueLoader;
const {ESBuildMinifyPlugin} = EsBuildLoader;
const {SourceMapDevToolPlugin} = webpack;
const __dirname = dirname(fileURLToPath(import.meta.url));
```

re-order just for the looks

**Comment by jolheiser** [comment]
fwiw I ran into this exact error yesterday.
I was using node 12.18, updating to 12.22 resolved the error for whatever reason.

If this fix is more permanent, that's great. 🙂

**Comment by silverwind** [comment]
@jolheiser good to know. Our minimum Node version is 12.17.0 so we should either require the higher version that fixed these resolutions or make them compatible with 12.17.0. I will check later.

**Comment by silverwind** [comment]
Looks like 12.20 is the version with the fix:

https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V12.md#2020-11-24-version-12200-erbium-lts-mylesborins

> (SEMVER-MINOR) named exports for CJS via static analysis (Guy Bedford) https://github.com/nodejs/node/pull/35249

**Comment by silverwind** [comment]
I think we can go with this fix for broader node compatibilty. Happy to approve after the suggestion above is merged.

**Comment by silverwind** [comment]
Can someone with push rights merge above suggestion so I can approve? :)

**Review by silverwind — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
does this need backporting?

**Comment by silverwind** [comment]
no, esm is only on master.

**Comment by techknowlogick** [comment]
🚀

**Pull request merged**
